### PR TITLE
support deno

### PR DESCRIPTION
**Pull request by vixalien**
**What**:

This PR allows `stylus` to work on `deno`, an alternative to Node.JS

**Why**:

Deno is gaining traction, and I personally use stylus, so I wanted to contrinue using stylus with Deno (after I switched to it from Node)

**How**:

The fix is two-steps:

1. Use `Object.setPrototypeOf` instead of the deprecated `__proto__` (which Deno doesn't support)
2. Fix a call in the `boolean` node that called the `Boolean` function, causing deno to run into an infinite loop as it was calling the constructor from the constructor in a recursive way.

**I THINK THIS IS OKAY, BUT PLEASE VERIFY**

**Checklist**:

- [ ] Documentation N/A
- [ ] Unit Tests N/A
- [x] Code complete
- [ ] Changelog

May I recommend moving to ES6 classes (would help) in the future, instead of modifying the prototype chain manually as this results in code reliability & performance issues.

### CODE REVIEW

**Comment by vixalien** [comment]
I forgot to add that this is a very minor change (in face of the user) as the API is unmodified and everything (including tests) works exactly as expected.

**Comment by vixalien** [comment]
anyone?

**Comment by iChenLei** [comment]
Can you add github action ci for stylus with deno environment ?

**Comment by vixalien** [comment]
@iChenLei  ptal

File: lib/nodes/object.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ var Object = module.exports = function Object(){
  * Inherit from `Node.prototype`.
  */
 
-Object.prototype.__proto__ = Node.prototype;
+globalThis.Object.setPrototypeOf(Object.prototype, Node.prototype);
```
Comment (iChenLei) [inline lib/nodes/object.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis#browser_compatibility See `globalThis ` compatibility

**Review by iChenLei — CHANGES_REQUESTED** [review]

**Comment by vixalien** [comment]
@iChenLei fixed

**Comment by vixalien** [comment]
@iChenLei I believe stylus's dependencies do not support older versions of node anymore.

In the logs you can see `mocha` says

```
unsupported engine for mocha@9.2.2: wanted: {"node":">= 12.0.0"} (current: {"node":"10.24.1","npm":"6.14.12"})
```

also `@adobe/css-tools` causes the test to fail since it uses an optional chaining operator (`?.`) which works on node v14+.

The `mocha` warning can be ignored. but the `@adobe/css-tools` one cannot be ignored. we'll need to use an older version of this package if supporting older versions of node is a must.

**Comment by vixalien** [comment]
@iChenLei ptal

**Comment by iChenLei** [comment]
I have a question: why not directly convert `function` into `class` (only needing [Node.js v4+](https://node.green/#ES2015-functions-class)), instead of using `setPrototypeOf`? Plus, you don't need to care about the issues with `@adobe/css-tools`.

**Review by iChenLei — COMMENTED** [review]
I have a question: why not directly convert `function` into `class` (only needing [Node.js v4+](https://node.green/#ES2015-functions-class)), instead of using `setPrototypeOf`? Plus, you don't need to care about the issues with `@adobe/css-tools`.

**Comment by vixalien** [comment]
that can be done yes, I was surprised why stylus does this too. since it'll be a huge change, I'll be working on a separate PR. stay tuned

**Comment by vixalien** [comment]
@iChenLei made a PR for Classes in #2813

**Comment by iChenLei** [comment]
So, can we close this PR ?

**Comment by vixalien** [comment]
yes.

**Pull request closed without merging**